client/webserver/webserver.go

```diff
@@ -659,6 +667,43 @@ func (s *WebServer) Addr() string {
 
 // Connect starts the web server. Satisfies the dex.Connector interface.
 func (s *WebServer) Connect(ctx context.Context) (*sync.WaitGroup, error) {
```

Comment:
@buck54321 do you have a suggestion how to fix this race?

```
==================
WARNING: DATA RACE
Read at 0x000001b81a10 by goroutine 20:
  decred.org/dcrdex/client/webserver.(*WebServer).Connect.func5()
      /home/user/dcrdex/client/webserver/webserver.go:759 +0xea
  decred.org/dcrdex/client/webserver.(*WebServer).Connect.func7()
      /home/user/dcrdex/client/webserver/webserver.go:766 +0x4f

Previous write at 0x000001b81a10 by goroutine 18:
  decred.org/dcrdex/client/webserver.New()
      /home/user/dcrdex/client/webserver/webserver.go:291 +0x73
  decred.org/dcrdex/client/webserver.TestConnectBindError()
      /home/user/dcrdex/client/webserver/webserver_test.go:518 +0x298
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:1595 +0x238
  testing.(*T).Run.func1()
      /usr/local/go/src/testing/testing.go:1648 +0x44

Goroutine 20 (running) created at:
  decred.org/dcrdex/client/webserver.(*WebServer).Connect()
      /home/user/dcrdex/client/webserver/webserver.go:758 +0x11ae
  decred.org/dcrdex/dex.(*ConnectionMaster).Connect()
      /home/user/dcrdex/dex/runner.go:122 +0xab
  decred.org/dcrdex/client/webserver.newTServer()
      /home/user/dcrdex/client/webserver/webserver_test.go:426 +0x38f
  decred.org/dcrdex/client/webserver.TestConnectBindError()
      /home/user/dcrdex/client/webserver/webserver_test.go:514 +0x44
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:1595 +0x238
  testing.(*T).Run.func1()
      /usr/local/go/src/testing/testing.go:1648 +0x44

Goroutine 18 (running) created at:
  testing.(*T).Run()
      /usr/local/go/src/testing/testing.go:1648 +0x82a
  testing.runTests.func1()
      /usr/local/go/src/testing/testing.go:2054 +0x84
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:1595 +0x238
  testing.runTests()
      /usr/local/go/src/testing/testing.go:2052 +0x896
  testing.(*M).Run()
      /usr/local/go/src/testing/testing.go:1925 +0xb57
  decred.org/dcrdex/client/webserver.TestMain.func1()
      /home/user/dcrdex/client/webserver/webserver_test.go:477 +0x58
  decred.org/dcrdex/client/webserver.TestMain()
      /home/user/dcrdex/client/webserver/webserver_test.go:479 +0x13b
  main.main()
      _testmain.go:89 +0x307
```

https://github.com/decred/dcrdex/actions/runs/8536376840/job/23384757884?pr=2715